Parser/string_parser.c

```diff
@@ -598,11 +370,21 @@ fstring_compile_expr(Parser *p, const char *expr_start, const char *expr_end,
         return NULL;
     }
 
-    str[0] = '(';
+    // Having curly braces at the beginning and at the end is needed so that the call
+    // to strstr in fstring_find_expr_location works correctly.
```

Comment:
Apologies if I am being too insistent on this, but I think it would be better if we clarify here what "works correctly" means, because I am afraid that I do understand the sentence because I already understand the problem, but for someone new to this and reading the code is unclear what the problem is or what "works correctly" means. (I talk also a higher level from just 'it will fail the sub-string match').

Reply:
No need to apologize. You're totally right that a newcomes would have to search around to understand this comment, which is at least sub-optimal. I'll try to come up with something more helpful.

Reply:
@pablogsal How does the new version look?

Reply:
Fantastic!